### PR TITLE
Move more components to plugins

### PR DESCRIPTION
**Pull request by mcoughlin**
This PR moves more components to plugins.

### CODE REVIEW

**Comment by stefanv** [comment]
I thought the plugins were going to be copied over during Fritz build, but here they are a PR onto SP itself? Functionality in SP itself does not need to be plugins.

**Comment by mcoughlin** [comment]
@stefanv the idea is that there are copied components between SP and the extensions. This factors out the extension code to just have the extra required code, rather than copying the whole thing.

**Comment by stefanv** [comment]
I'm still not following. Presumably Fritz-specifics would be plugins provided by Fritz, that get copied to SkyPortal.

**Comment by mcoughlin** [comment]
@stefanv yes exactly. We are just copying less code.

**Comment by stefanv** [comment]
Right, so why are these plugins submitted as PRs to SkyPortal?

**Comment by mcoughlin** [comment]
@stefanv well still need to open a placeholder for the code in the main file.

**Comment by stefanv** [comment]
But the default placeholder will presumably be empty, as it is elsewhere. Why populate it with Fritz specifics?

**Comment by mcoughlin** [comment]
@stefanv The placeholders are empty, best I can tell. Which ones do you mean?

**Comment by stefanv** [comment]
I see a bunch of images committed; so you mean to include the authors in this commit?

If so, is the intent to be able to override the SP authors list in Fritz?

**Comment by mcoughlin** [comment]
@stefanv I assumed the lack of authors was simply an oversight appearing in fritz only; I am pretty sure it belongs in SP.

**Comment by stefanv** [comment]
I think the distinction to think through is: what content do we want to make *removable*, and what content should always be there.  Plugins add to pages, they cannot subtract.  So, if you want people to be able to completely rewrite the About page, the current approach makes sense.  If you think the authors + citation should always be there, with additional content added as a plugin, then this needs to be modified to have authors + citations in the About page itself.

**Comment by mcoughlin** [comment]
@stefanv At this point, I have the authors in the About page, the citations are instance specific so are in the plugin.

**Comment by stefanv** [comment]
:+1: Sounds good.

Let's get those images small; some of them are 150K, and they should be about 10K each.

You can use `convert` to make them smaller, e.g.:

```
convert x.png -resize 300x y.png
```

(300 is just a guess, whatever size we need the images to be at)

**Comment by stefanv** [comment]
(One could argue that at least the SP citation should be listed; we want people to cite the package, don't we?)

**Comment by mcoughlin** [comment]
@stefanv I had that in the Plugin due to ease of how different the Fritz and SkyPortal discussion was (including citation list).

**Review by guynir42 — APPROVED** [review]
Minor comments. Looks good overall.

File: static/js/components/CentroidPlot.jsx [hunk 1]
```diff
@@ -59,7 +78,7 @@ const getCirclePoints = (delRaGroup, delDecGroup) => {
   const medianRA = d3.median(delRaGroup);
   const medianDec = d3.median(delDecGroup);
 
-  const points = thetas?.map((theta) => {
+  const points = thetas.map((theta) => {
```
[the remark this answers is not in the document]
Comment (mcoughlin) [reply static/js/components/CentroidPlot.jsx:81]: Oops.

File: static/js/components/CentroidPlot.jsx [hunk 2]
```diff
@@ -206,83 +229,152 @@ const spec = (inputData, textColor, titleFontSize, labelFontSize) => ({
...
-          field: "filter",
-          type: "nominal",
-          scale: { range: ["circle", "square", "triangle"] },
-        },
-        size: { value: 35 },
+        // shape: {
+        //   field: "filter",
+        //   type: "nominal",
+        //   scale: { range: ["circle", "square", "triangle"] },
+        // },
```
[the remark this answers is not in the document]
Comment (mcoughlin) [reply static/js/components/CentroidPlot.jsx:249]: yup.

**Pull request merged**